tests/UI/pages/FO/FObasePage.js

```diff
@@ -313,18 +316,43 @@ class FOBasePage extends CommonPage {
     await this.clickAndWaitForNavigation(page, this.cartLink);
   }
 
+  /**
+   * Check if there are autocomplete search result
+   * @param page {Page} Browser tab
+   * @param productName {string} Product name to search
+   * @returns {Promise<boolean>}
+   */
+  async hasAutocompleteSearchResult(page, productName) {
+    await this.setValue(page, this.searchInput, productName);
+    await page.waitForTimeout(2000);
+    return this.elementVisible(page, this.autocompleteSearchResult, 2000);
+  }
+
   /**
    * Get autocomplete search result
    * @param page {Page} Browser tab
    * @param productName {string} Product name to search
-   * @returns {Promise<*>}
+   * @returns {Promise<string>}
    */
   async getAutocompleteSearchResult(page, productName) {
     await this.setValue(page, this.searchInput, productName);
     await page.waitForTimeout(2000);
     return this.getTextContent(page, this.autocompleteSearchResult);
   }
 
+  /**
+   * Count autocomplete search result
+   * @param page {Page} Browser tab
+   * @param productName {string} Product name to search
+   * @returns {Promise<int>}
+   */
+  async countAutocompleteSearchResult(page, productName) {
+    await this.setValue(page, this.searchInput, productName);
+    await page.waitForTimeout(2000);
+    await this.waitForVisibleSelector(page, this.autocompleteSearchResultItem);
+    return (await page.$$(this.autocompleteSearchResultItem)).length;
```

Comment:
```suggestion
    return page.$$eval(this.autocompleteSearchResultItem, all => all.length);
```
you can avoid a double promise like this.

Reply:
Fixed